initfs/scripts/moslo-build.sh

```diff
@@ -108,17 +134,26 @@ add_dependencies()
                 fi
 
                 #Use ld-linux for libraries
+                LD_LINUX=$LD
+
                 $($LD --verify $i)
                 if [ "$?" -ne "2" ] ; then
-                        LD_LINUX=""
+                        #Get direct dependencies
+                        DIRECT_LIBS=$(objdump_find_lib_path $i)
+                        DEP="$DIRECT_LIBS"
+                        LD_LINUX="ldd"
+                        #Get full dependencies of the direct dependencies
+                        for lib in $DIRECT_LIBS; do
+                                TEMPDEPS=$($LD_LINUX $lib \
+                                        | sed -ne "s/.*[\t ]\(\/.*\) (.*/\1/gp")
+                                DEP="$DEP $TEMPDEPS"
+                        done
                 else
-                        LD_LINUX=$LD
+                        #Get dependencies
+                        DEP=$($LD_LINUX $i \
+                                | sed -ne "s/.*[\t ]\(\/.*\) (.*/\1/gp")
```

Comment:
Sounds reasonable idea. well named function would actually tell what it does. You never really know that from looking at the sed line. Now the problem is just finding out what it does, as I copied it from the old code :P 

I'll fix something for it.